File: java/src/main/java/io/fivexer/sdk/model/BulkTaskResult.java
package io.fivexer.sdk.model;

/**
 * One entry of a bulk create. The wire sends a union — either an id or an error — so exactly one of {@code id}/{@code error} is set; {@link #isOk()} is the discriminator.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class BulkTaskResult {
    private Integer index;
    private String id;
    private String status;
    private BulkTaskError error;

    public Integer getIndex() { return index; }
    public String getId() { return id; }
    public String getStatus() { return status; }
    public BulkTaskError getError() { return error; }

    /** True when this entry was created; false when it carries an error instead. */
    public boolean isOk() { return error == null; }
}

File: php/src/Model/BulkTaskResult.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

/**
 * One entry of a bulk create. The wire sends a union — either an id or an error — so exactly one of `id`/`error` is set; {@see self::ok()} is the discriminator.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class BulkTaskResult
{
    public function __construct(
        /** Position in the caller's input list. */
        public readonly int $index,
        public readonly ?string $id,
        public readonly ?string $status,
        public readonly ?BulkTaskError $error,
    ) {
    }

    /** True when this entry was created; false when it carries an error instead. */
    public function ok(): bool
    {
        return $this->error === null;
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            index: (int) ($data['index'] ?? 0),
            id: isset($data['id']) ? (string) $data['id'] : null,
            status: isset($data['status']) ? (string) $data['status'] : null,
            error: isset($data['error']) && \is_array($data['error']) ? BulkTaskError::fromArray($data['error']) : null,
        );
    }
}
